src/main/scala/tilelink/Parameters.scala

```diff
@@ -1085,27 +1157,73 @@ case class TLEdgeParameters(
+  def expectsPutFullMasterToSlaveCheckerSafe   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartialChecker(sourceId, lgSize, range)  && slave.supportsPutPartialCheckerSafe(address, lgSize, range)
+  def expectsPutPartialMasterToSlaveCheckerSafe(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartialChecker(sourceId, lgSize, range)  && slave.supportsPutPartialCheckerSafe(address, lgSize, range)
+  def expectsHintMasterToSlaveCheckerSafe      (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsHintChecker(sourceId, lgSize, range)        && slave.supportsHintCheckerSafe(address, lgSize, range)
+
+  def expectsAcquireTMasterToSlaveCheckerFast  (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsAcquireTChecker(sourceId, lgSize, range)    && slave.supportsAcquireTCheckerFast(address, lgSize, range)
+  def expectsAcquireBMasterToSlaveCheckerFast  (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsAcquireBChecker(sourceId, lgSize, range)    && slave.supportsAcquireBCheckerFast(address, lgSize, range)
+  def expectsArithmeticMasterToSlaveCheckerFast(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsArithmeticChecker(sourceId, lgSize, range)  && slave.supportsArithmeticCheckerFast(address, lgSize, range)
+  def expectsLogicalMasterToSlaveCheckerFast   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsLogicalChecker(sourceId, lgSize, range)     && slave.supportsLogicalCheckerFast(address, lgSize, range)
+  def expectsGetMasterToSlaveCheckerFast       (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsGetChecker(sourceId, lgSize, range)         && slave.supportsGetCheckerFast(address, lgSize, range)
+  def expectsPutFullMasterToSlaveCheckerFast   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutFullChecker(sourceId, lgSize, range)     && slave.supportsPutFullCheckerFast(address, lgSize, range)
+  def expectsPutPartialMasterToSlaveCheckerFast(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartialChecker(sourceId, lgSize, range)  && slave.supportsPutPartialCheckerFast(address, lgSize, range)
+  def expectsHintMasterToSlaveCheckerFast      (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsHintChecker(sourceId, lgSize, range)        && slave.supportsHintCheckerFast(address, lgSize, range)
+
+  //We don't use Safe vs Fast because we don't have the equivalent of address decoder for sourceIds
+  //TODO (can use sourceId here to see if the specific master can support this transaction)
```

Comment:
Please do this TODO, i.e. add the arguments to make the below SlaveToMaster functions match the above MasterToSlave functions, and cal the `slave.emitsXChecker` function using some of those arguments.